lib/active_model/serializer/associations.rb

```diff
@@ -48,7 +48,13 @@ def option(key, default=nil)
 
         def target_serializer
           serializer = option(:serializer)
-          serializer.is_a?(String) ? serializer.constantize : serializer
+          if serializer.is_a?(String)
+            serializer.constantize
+          elsif serializer.is_a?(Symbol)
+            source_serializer.send(serializer)
+          else
+            serializer
```

Comment:
Interesting point. I'm not really sure what I think about it. 

This code could be

```ruby
case serializer
when String then serializer.constantize
when Symbol then source_serializer.send(serializer)
else serializer
end
```